File: 프로그래머스/lv2/42747. H－Index/H－Index.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> citations) {
    int n = citations.size();
    for (int h = n; h > 0; h--) {
        int cnt = 0;
        for (int cit : citations)
            if (cit >= h) cnt++;
        if (cnt >= h) return h;
    }
    return 0;
}

File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
        Integer[] arr = Arrays.stream(citations).boxed().toArray(Integer[]::new);
        Arrays.sort(arr, Collections.reverseOrder());
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] <= i + 1) return i;
        }
        return citations.length;
    }
}